### PR TITLE
Add bqrs types and ability to run template queries.

### PR DESCRIPTION
**Pull request by jcreedcmu**
These are not-yet-user facing changes required by https://github.com/github/codeql-coreql-team/issues/28

### CODE REVIEW

**Comment by aeisenberg** [comment]
I can't review the `bqrs-cli-types.ts` file, but everything else looks good.

File: extensions/ql-vscode/src/bqrs-cli-types.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  startColumn: never,
+  endLine: never,
+  endColumn: never,
+  charOffset: number,
+  charLength: number,
+}
+
+export interface WholeFileLocation {
+    uri: string,
+    startLine: undefined,
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/bqrs-cli-types.ts]: nit: should these be `never` as well?

Comment (jcreedcmu) [reply extensions/ql-vscode/src/bqrs-cli-types.ts]: Huh, it seems like they should be. @alexet  was there some reason these had to be `undefined`?

File: extensions/ql-vscode/src/bqrs-cli-types.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+export type ColumnKind = "f" | "i" | "s" | "b" | "d" | "e";
+
+export interface Column {
+  name?: string,
+  kind: ColumnKind,
+}
+
+
+export interface ResultSetSchema {
```
[the remark this answers is not in the document]
Comment (jcreedcmu) [reply extensions/ql-vscode/src/bqrs-cli-types.ts]: Not a bad suggestion, but I prefer leaving this the way it is for now for the reason we should be able to move off of `lib/semmle-bqrs` anyhow.

File: extensions/ql-vscode/src/cli.ts [hunk 3]
```diff
@@ -471,6 +472,36 @@ export class CodeQLCliServer implements Disposable {
...
+      pageSize ? ["--paginate-rows", pageSize.toString()] : []
+    ).concat(
+      bqrsPath
+    );
+    return await this.runJsonCodeQlCliCommand<BQRSInfo>(['bqrs', 'info'], subcommandArgs, "Reading bqrs header");
+  }
+
+  /**
+  * Gets the results from a bqrs.
+  * @param config The configuration containing the path to the CLI.
```
[the remark this answers is not in the document]
Comment (jcreedcmu) [reply extensions/ql-vscode/src/cli.ts]: Ah, good catch.

File: extensions/ql-vscode/src/bqrs-cli-types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (aeisenberg) [reply extensions/ql-vscode/src/bqrs-cli-types.ts]: Fair enough.

File: extensions/ql-vscode/src/bqrs-cli-types.ts (hunk 1 above)
Comment (jcreedcmu) [reply extensions/ql-vscode/src/bqrs-cli-types.ts]: Doesn't break the build; Just going to change them optimistically.

**Pull request merged**